### ISSUE TITLE
binary install script not working for some releases

### ISSUE THREAD

**Issue by radTuti** [kubernetes-sigs/kustomize#3154]
**Describe the bug**

install kustomize script does not appear to work for latest version (v3.8.6) when specified or when no version string is specified 

**Files that can reproduce the issue**

[install_kustomize.sh](https://github.com/kubernetes-sigs/kustomize/blob/6a9e75ee0d558683b989fe42e7edc02e84c990a9/hack/install_kustomize.sh)

**Expected/Actual output**

- using script without specifying a version
```sh
$ curl -sfL https://raw.githubusercontent.com/\
> kubernetes-sigs/kustomize/master/hack/install_kustomize.sh  | bash
Error: kustomize binary with the version  does not exist!
```

- using script specifying the latest (3.8.6) version
```sh
$ bash <(curl -sfL https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh) 3.8.6
Error: kustomize binary with the version 3.8.6 does not exist!
```

- using script with a previous version (3.8.5)
```sh
$ bash <(curl -sfL https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh) 3.8.5
{Version:kustomize/v3.8.5 GitCommit:4052cd4fd8c76a17b5f64e32509f3fba9713fe75 BuildDate:2020-10-08T02:45:59Z GoOs:linux GoArch:amd64}
kustomize installed to current directory.
```

**Kustomize version**

N/A

**Platform**

Linux

**Additional context**

N/A

**Comment by Shell32-Natsu** [comment]
This is because we have added ARM linux releases in the latest release. Will fix it soon.

**Comment by radTuti** [comment]
@Shell32-Natsu thanks! i created a stop-gap change & PR to get the script working as expected. Not sure if the script also needs to support ARM linux releases

**Comment by mhamann** [comment]
You can also change the install command to `curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s 3.8.5` as a workaround.

Probably should've rolled back / removed the ARM release until this was fixed though. Pretty sure there are a million builds across the planet that are failing right now because of this. Basically affects every linux on amd64 install out there.

**Comment by Shell32-Natsu** [comment]
The fix has been merged in #3155. Please check.

**Comment by Shell32-Natsu** [comment]
If you are using https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh, it may take several minutes to update the cache.
